rclpy/src/rclpy/_rclpy.c

```diff
@@ -299,68 +601,99 @@ rclpy_wait_set_init(PyObject * Py_UNUSED(self), PyObject * args)
   Py_RETURN_NONE;
 }
 
-/// Clear all the subscription pointers in the waitset
+/// Clear all the pointers of a given wait_set field
 /*
+ * \param[in] entity_type string defining the entity ["subscription, client, service"]
  * \param[in] pywait_set Capsule pointing to the waitset structure
  * \return NULL
  */
 static PyObject *
-rclpy_wait_set_clear_subscriptions(PyObject * Py_UNUSED(self), PyObject * args)
+rclpy_wait_set_clear_entities(PyObject * Py_UNUSED(self), PyObject * args)
 {
+  const char * entity_type;
   PyObject * pywait_set;
 
-  if (!PyArg_ParseTuple(args, "O", &pywait_set)) {
+  if (!PyArg_ParseTuple(args, "zO", &entity_type, &pywait_set)) {
     return NULL;
   }
 
   rcl_wait_set_t * wait_set = (rcl_wait_set_t *)PyCapsule_GetPointer(pywait_set, NULL);
-  rcl_ret_t ret = rcl_wait_set_clear_subscriptions(wait_set);
+  rcl_ret_t ret;
+  if (0 == strcmp(entity_type, "subscription")) {
+    ret = rcl_wait_set_clear_subscriptions(wait_set);
+  } else if (0 == strcmp(entity_type, "client")) {
+    ret = rcl_wait_set_clear_clients(wait_set);
+  } else if (0 == strcmp(entity_type, "service")) {
+    ret = rcl_wait_set_clear_services(wait_set);
+  } else {
+    PyErr_Format(PyExc_RuntimeError,
+      "%s is not a known entity", entity_type);
+    Py_RETURN_FALSE;
+  }
   if (ret != RCL_RET_OK) {
     PyErr_Format(PyExc_RuntimeError,
-      "Failed to clear subscriptions from wait set: %s", rcl_get_error_string_safe());
-    return NULL;
+      "Failed to %s from waitset: %s", entity_type, rcl_get_error_string_safe());
```

Comment:
The placeholders should be quoted if they appear in the text, e.g.:

> "Failed to '%s' from waitset: %s"

Same below.